CMakeLists.txt

```diff
@@ -27,6 +27,8 @@ set(PLUGIN_VERSION_MAJOR 0)
 set(PLUGIN_VERSION_MINOR 0)
 set(PLUGIN_VERSION_BUILD 1)
 
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
Enforcing a new language standard just for one function? Seriously?

Reply:
Seriously. I don't see any reason not to. std::string_view - very good implementation string. It structure of a pointer to a string and its length. This way I can use: string (char) / std::wstring / std::string. In doing so, I need to overload the function for each implementation.

Reply:
This plugin template uses the C++03 standard (at least that's the intention), so the user could choose between that one, C++11 or C++17, without being forced to use any of the newer standards, and I'm not planning to change this.

Reply:
Anyway, I see what you mean: you want to use `std::string_view` which has the `data()` method, but doesn't have `c_str()`. This can be resolved with conditional compilation, without enforcing the C++17 standard:
```C++
#if __cplusplus >= 201703L // TODO: Also check _MSC_VER, because Microsoft
    // use the `data()` method
#else
    // use the `c_str()` method
#endif
```